### ISSUE TITLE
PURL Format Discrepancy

### ISSUE THREAD

**Issue by kashishtopi** [DependencyTrack/dependency-track#2694]
### Current Behavior

![image](https://user-images.githubusercontent.com/46081558/233087782-3012c713-5bec-4aba-9e07-f039534377db.png)
![image](https://user-images.githubusercontent.com/46081558/233087856-d0d931d5-c32f-4a73-bbe5-e23ff07ac85e.png)
![image](https://user-images.githubusercontent.com/46081558/233088075-6ca92442-e807-46d9-b95c-6ac7ecaa47f3.png)

Although, The Purl format on [Dependency Track](https://docs.dependencytrack.org/datasources/routing/) and [Purl-Spec](https://github.com/package-url/purl-spec) mention that Namespace/GroupID is optional.

For JAVA (Maven) packages, Dependency Track fails to find vulnerabilities in the Java Maven SBOM If `Namespace/GroupID` is not mentioned in the Purl.

### Proposed Behavior

scheme:type/namespace/name@version

** Namespace is not optional and mandatory in the Purl-Spec format for being detectable by Dependency Track.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

**Comment by melba-lopez** [comment]
hey team!  if you need more info on this, feel free to reach out to myself or @kashishtopi as we have been troubleshooting this issue for a few days now.

**Comment by stevespringett** [comment]
> Namespace is not optional, IT IS a MUST in the Purl-Spec format for a JAVA - Maven SBOM for being detectable by Dependency Track.

That is correct and is documented in the purl spec. See https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst

I do not believe it is the role of Dependency-Track to validate all current and future purl types.

**Comment by stevespringett** [comment]
> we would want to inform the users that there may be reasons why they see NO vulnerabilities

What you're describing is a subset of SBOM quality. Quality is being discussed by the larger industry from two angles: taxonomy/capabilities and accuracy. Accuracy is a lot more difficult to analyze. Purl being just one aspect of component identity. CPE, SWID, and coordinates (GAV/GNV) being the others. Validating for one without validating against the others will have limited utility.

Checking for the existence of all required fields for a given ecosystem doesn't increase quality that much. You'd have to go much further in verifying the existence of the package itself based on the data in the SBOM. Otherwise, the values could simply be made up. If a group/namespace is missing for a Maven package, that's an indication that the tool or method that produced the SBOM should likely not be trusted. That would be good to flag. But if we go down the rabbit hole of individual component SBOM quality, it's going to involve much more than just purl validation.

**Comment by stevespringett** [comment]
If you wanted to focus specifically on purl, I'd recommend adding purl type checks to the official purl implementations for Java, go, .NET, etc.

https://github.com/orgs/package-url/repositories
